assets/src/dashboard/app/views/shared/storyGridView.js

```diff
@@ -143,22 +90,27 @@ const StoryGridView = ({
               <CardTitle
                 title={story.title}
                 status={story?.status}
+                id={story.id}
                 secondaryTitle={
                   isSavedTemplate
                     ? __('Google', 'web-stories')
                     : users[story.author]?.name
                 }
                 displayDate={story?.modified}
-                editMode={titleRenameId === story.id}
-                onEditComplete={(newTitle) =>
-                  handleOnRenameStory(story, newTitle)
-                }
-                onEditCancel={() => setTitleRenameId(-1)}
+                {...(renameStory
+                  ? {
+                      editMode: renameStory?.id === story?.id,
+                      onEditComplete: (newTitle) =>
+                        renameStory?.handleOnRenameStory(story, newTitle),
+                      onEditCancel: renameStory?.handleCancelRename,
+                    }
+                  : {})}
```

Comment:
I did this because I believe we can't rename saved templates - at least definitely not right now. By moving the functionality of the story menu up a level I had to adjust the saved template usage of this shared component, while doing that I gutted the story specific functionality since it's going to be different for templates.

Reply:
This is a bit confusing structure wise. I think we could turn this into an object above and then spread the appropriate thing.